src/main.ts

```diff
@@ -65,6 +65,38 @@ const createBranch = async (
 	}
 };
 
+/** Merges "head" into "base" on the given owner/repo.
+ *
+ * Returns true if a merge commit was created, otherwise false
+ *
+ * Status code reference: https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#merge-a-branch--status-codes
+ */
+const merge = async (
+	okit: Octokit,
+	{ owner, repoName, base, head }: { owner: string; repoName: string; base: string; head: string },
+): Promise<boolean> => {
+	core.debug(`Will attempt to merge ${head} into ${base}`);
+	const { status } = await okit.repos.merge({
+		owner,
+		repo: repoName,
+		base,
+		head,
+	});
+
+	if (status === 201) {
+		core.info(`Merged ${head} into ${base}`);
+		return true;
+	}
+
+	if (status === 204) {
+		core.info(`${head} is already merged to ${base}`);
+		return false;
+	}
+
+	core.error(`Unknown return status (${status}). Assuming we don't need to kick CI.`);
+	return false;
+};
```

Comment:
Do 5XX responses throw? 3XX? Instead of ret false, should we throw?

Part of me doesn't like the three results of this func call: a throw (assuming from the catches), ret false, and ret true. Are all three cases necessary?

Reply:
> Do 5XX responses throw? 3XX? Instead of ret false, should we throw?

The client will throw, yeah. The last case there is essentially an impossible case. If we get some other 2xx return code, it's undocumented.

> Are all three cases necessary?

Yeah, so I need to know...

- Merge success (true)
- Merge wasn't required (false)
- Merge failed (due to perms, merge conflict, etc) (throw) 

I oughta update the docs to explain that the fn can throw